obj/empty.go

```diff
@@ -21,22 +17,15 @@ func isEmptyValue(v reflect.Value) bool {
 		return v.Float() == 0
 	case reflect.Interface, reflect.Ptr:
 		return v.IsNil()
+	case reflect.Struct:
+		// Note: we can't rely on a *literal* "is zero" check because
+		// non-zero pointers may still point to *empty* things.
+		for i := 0; i < v.NumField(); i++ {
+			if !isEmptyValue(v.Field(i)) {
+				return false
+			}
+		}
+		return true
 	}
 	return false
 }
-
-// zeroishness checks on a struct are possible in theory but we're passing for now.
-// There's no easy route exposed from the reflect package, due to a variety of reasons:
-//
-//   - The `DeepEqual` method would almost suffice, but we'd need one that takes
-//     `reflect.Value` instead of `interface{}` params, because we already have
-//     the former, and un/re-boxing those into two `interface{}` values is two
-//     heap mallocs and that's a wildly unacceptable performance overhead for this.
-//     `DeepEqual` calls `deepValueEqual`, which does what we want, but...
-//   - We can't easily copy the `deepValueEqual` method.  It imports the unsafe
-//     package.  This is undesirable because it would reduce the portability of refmt.
```

Comment:
Unfortunately, these statements aren't *quite* correct for the reason I stated in my comment: we care about *empty*, not *zero*. We can (and often do) have non-zero "empty" values.